src/CondenserDotNet.Server/IConsulService.cs

```diff
@@ -4,6 +4,7 @@ namespace CondenserDotNet.Server
 {
     public interface IConsulService : IService
     {
-        Task Initialise(string serviceId, string nodeId, string[] tags, string address, int port);
+        Task Initialise(string serviceId, string nodeId, string[] tags, string address, int port,
+            ICurrentState stats);
```

Comment:
If we are resolving the class from the DI container, should we move the stats to the constructor and get the injecction for free?